cinnamonvantage@garlayntoji/files/cinnamonvantage@garlayntoji/applet.js

```diff
@@ -0,0 +1,249 @@
+// This extension was developed by:
+// * Garlayn Toji
+// Version: 1.0.0
+// Date: 2024-08-07
+// License: GPLv2+
+// Copyright (C) 2024-2025 Garlayn Toji
+// Cinnamon Vantage Cinnamon Applet
+
+const Lang = imports.lang;
+const Applet = imports.ui.applet;
+const PopupMenu = imports.ui.popupMenu;
+const Util = imports.misc.util;
+const GLib = imports.gi.GLib;
+const Gio = imports.gi.Gio;
+const Mainloop = imports.mainloop;
+const Settings = imports.ui.settings;
+const Main = imports.ui.main;
+
+const UUID = "cinnamonvantage@garlayntoji";
+
+// Icon location
+const ICON_PATH = GLib.get_home_dir() + '/.local/share/cinnamon/applets/cinnamonvantage@garlayntoji/icon.png';
+
+// Schema keys
+const FN_LOCK = 'fnLock';
+const WINKEY = 'winkey';
+const TOUCHPAD = 'touchpad';
+const CONSERVATION_MODE = 'conservationMode';
+const RAPID_CHARGE = 'rapidCharge';
+const USB_CHARGING = 'usbCharging';
+const OVERDRIVE = 'overdrive';
+const GSYNC = 'gsync';
+
+// File paths
+const FILE_PATHS = {
+    [FN_LOCK]: "/sys/bus/platform/drivers/ideapad_acpi/VPC2004:00/fn_lock",
+    [CONSERVATION_MODE]: "/sys/bus/platform/drivers/ideapad_acpi/VPC2004:00/conservation_mode",
+    [USB_CHARGING]: "/sys/bus/platform/drivers/ideapad_acpi/VPC2004:00/usb_charging",
+    [WINKEY]: "/sys/bus/platform/drivers/legion/PNP0C09:00/winkey",
+    [TOUCHPAD]: "/sys/bus/platform/drivers/legion/PNP0C09:00/touchpad",
+    [RAPID_CHARGE]: "/sys/bus/platform/drivers/legion/PNP0C09:00/rapidcharge",
+    [OVERDRIVE]: "/sys/bus/platform/drivers/legion/PNP0C09:00/overdrive",
+    [GSYNC]: "/sys/bus/platform/drivers/legion/PNP0C09:00/gsync"
+};
+
+// Define Applet and Applet settings
+function MyApplet(metadata, orientation, panelHeight, instanceId) {
+    this.settings = new Settings.AppletSettings(this, UUID, instanceId);
+    this._fileMonitors = {};
+    this._init(metadata, orientation, panelHeight, instanceId);
+}
+
+MyApplet.prototype = {
+    __proto__: Applet.IconApplet.prototype,
+
+    _init: function(metadata, orientation, panelHeight, instanceId) {
+        Applet.IconApplet.prototype._init.call(this, orientation, panelHeight, instanceId);
+
+        try {
+            this.set_applet_icon_symbolic_path(ICON_PATH);
+            this.set_applet_tooltip(_("Cinnamon Vantage"));
```

Comment:
It doesn't appear that you have translation support implemented yet.

Reply:
Is `./cinnamon-spices-makepot UUID` used to generate a .pot file? Either way, when I execute this command, here's what I get:
```
Traceback (most recent call last):
  File "/home/hugoc/Documents/cinnamon-spices-applets/./cinnamon-spices-makepot", line 316, in <module>
    parse_args()
  File "/home/hugoc/Documents/cinnamon-spices-applets/./cinnamon-spices-makepot", line 62, in parse_args
    output = make_pot(args.uuid)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/hugoc/Documents/cinnamon-spices-applets/./cinnamon-spices-makepot", line 290, in make_pot
    with open(pot_file_path, 'r+', encoding='utf-8') as po:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/home/hugoc/Documents/cinnamon-spices-applets/cinnamonvantage@garlayntoji/files/cinnamonvantage@garlayntoji/po/cinnamonvantage@garlayntoji.pot'
```
The output will be the same whether a pot file is already present or not, as the script will delete the existing file.

Reply:
I can't replicate it locally. To be clear though, this script just generates the files for translations, it doesn't implement translation support for your applet.